kmod/patch/kpatch-patch-hook.c

```diff
@@ -303,38 +292,84 @@ static int patch_make_dynrelas_list(struct list_head *objects)
 	return 0;
 }
 
-static int patch_make_hook_lists(struct list_head *objects)
+static int patch_set_callbacks(struct list_head *objects)
 {
+	struct kpatch_pre_patch_callback *p_pre_patch_callback;
+	struct kpatch_post_patch_callback *p_post_patch_callback;
+	struct kpatch_pre_unpatch_callback *p_pre_unpatch_callback;
+	struct kpatch_post_unpatch_callback *p_post_unpatch_callback;
 	struct kpatch_object *object;
-	struct kpatch_patch_hook *p_hook;
-	struct kpatch_hook *hook;
 
-	for (p_hook = __kpatch_hooks_load; p_hook < __kpatch_hooks_load_end;
-	     p_hook++) {
-		object = patch_find_or_add_object(objects, p_hook->objname);
+	for (p_pre_patch_callback = __kpatch_callbacks_pre_patch;
+	     p_pre_patch_callback < __kpatch_callbacks_pre_patch_end;
+	     p_pre_patch_callback++) {
+
+		object = patch_find_or_add_object(objects, p_pre_patch_callback->objname);
 		if (!object)
 			return -ENOMEM;
 
-		hook = kzalloc(sizeof(*hook), GFP_KERNEL);
-		if (!hook)
+		if (object->pre_patch_callback) {
+			pr_err("extra pre-patch callback for object: %s\n",
+				object->name ? object->name : "vmlinux");
```

Comment:
Another place where object->name can never be NULL.  There are several more in this function.  It would be nice to make kpatch consistent with livepatch there, though maybe that's out of scope for this PR.